Tests/AVFoundationNowPlayingBuilderTests.swift

```diff
@@ -0,0 +1,257 @@
+                        let filteredItem = nowPlayingBuilder?.items.filter{ $0.identifier == AVMetadataCommonIdentifierDescription }
+                        expect(filteredItem).to(beEmpty())
+                    }
+                }
+            }
+
+            describe("#getDate()") {
+                context("When options has kMetaDataDate") {
+
+                    beforeEach {
+                        metadata = [kMetaDataDate: Date()]
+                        nowPlayingBuilder = AVFoundationNowPlayingBuilder(metadata: metadata!)
+                    }
+
+                    it("returns non nil value") {
+                        expect(nowPlayingBuilder?.getDate()).toNot(beNil())
+                    }
+
+                    it("sets the item to the list of items") {
+                        let filteredItem = nowPlayingBuilder?.items.filter{ $0.identifier == AVMetadataCommonIdentifierCreationDate }
+                        expect(filteredItem).toNot(beEmpty())
+                    }
+                }
+
+                context("When options hasn't kMetaDataDate") {
+
+                    beforeEach {
+                        metadata = [:]
+                        nowPlayingBuilder = AVFoundationNowPlayingBuilder(metadata: metadata!)
+                    }
+
+                    it("returns nil value") {
+                        expect(nowPlayingBuilder?.getDate()).to(beNil())
+                    }
+
+                    it("doesn't set the item to the list of items") {
+                        let filteredItem = nowPlayingBuilder?.items.filter{ $0.identifier == AVMetadataCommonIdentifierCreationDate }
+                        expect(filteredItem).to(beEmpty())
+                    }
+                }
+            }
+
+            describe("#getArtwork(with options)") {
+
+                context("When has MetaDataArtwork") {
+
+                    beforeEach {
+                        metadata = [kMetaDataArtwork: self.generateNewImage()]
+                        nowPlayingBuilder = AVFoundationNowPlayingBuilder(metadata: metadata!)
+                    }
+
+                    it("returns non nil value") {
+                        waitUntil() { done in
+                            nowPlayingBuilder?.getArtwork(with: [:]) { item in
+                                expect(item).toNot(beNil())
+                                done()
+                            }
+                        }
+                    }
+                }
+
+                context("When doesn't have metadata image") {
+
+                    beforeEach {
+                        metadata = [:]
+                        nowPlayingBuilder = AVFoundationNowPlayingBuilder(metadata: metadata!)
+                    }
+
+                    it("loads poster image") {
+                        waitUntil() { done in
+                            let url = "https://cloud.githubusercontent.com/assets/1156242/16349649/54f233e2-3a30-11e6-98e4-42eb5284b730.png"
```

Comment:
Network access shall be avoided in unit tests.

Reply:
Its necessary to add a framework to stub this request. Seems better to not add new dependencies on Clappr to avoid compatibility problems.

Do you have another suggestion for this problem?

Reply:
I consider that we can add this new framework as internal testing only.

Reply:
I agree with Fernando. I don't see any problem in adding internal frameworks.